mlrun/feature_store/api.py

```diff
@@ -99,6 +99,7 @@ def get_offline_features(
     update_stats: bool = False,
     engine: str = None,
     engine_args: dict = None,
+    query: str = None,
```

Comment:
Is it query or filter? The PR title says filter but the code says query.

Reply:
In the origin the param was filter as well, but @guy1992l  told me that filter is a reserved word in python, the purpose of this PR it to filter the offline feature that the user gets from `get_offline_features()` by using this string query

Reply:
@Tankilevitch

Reply:
I think it should be changed to query (as seen in `pandas.DataFrame` for example) or at least to `fmt` as format is a reserved word in python. But, you shouldn't change API like this without backward compatibility, either do not change it now, or change it but leave the `filter` keyword and simply have:

```
if filter is not None:
    warnings.warn("This argument is renamed to 'query' so use `query` instead.", PendingDeprecationWarning)
    query = filter
``` 

In my opinion, we can skip this adventure for now, Yaron do not like this type of changes (unless he told you to do so :))

Reply:
I'm ok with `query`

Reply:
I'm fine with query as well, but please be consistent and change filter to query everywhere, including the comments, test name, and PR title and description.

Reply:
@gtopper  - I changed all the relevant  places, and I mention that the filter performed by the query argument.